pkg/testsuite/mock/wallet_transactions.go

```diff
@@ -0,0 +1,386 @@
+	))
+
+	// register the outputs in the wallet
+	w.registerOutputs(transactionName, signedTransaction.Transaction)
+
+	return signedTransaction
+}
+
+// DelayedClaimingTransition transitions DelegationOutput into delayed claiming state by setting DelegationID and EndEpoch.
+func (w *Wallet) DelayedClaimingTransition(transactionName string, inputName string, creationSlot iotago.SlotIndex, delegationEndEpoch iotago.EpochIndex) *iotago.SignedTransaction {
+	input := w.Output(inputName)
+	if input.OutputType() != iotago.OutputDelegation {
+		panic(ierrors.Errorf("%s is not a delegation output, cannot transition to delayed claiming state", inputName))
+	}
+
+	prevOutput, ok := input.Output().Clone().(*iotago.DelegationOutput)
+	if !ok {
+		panic(ierrors.Errorf("cloned output %s is not a delegation output, cannot transition to delayed claiming state", inputName))
+	}
+
+	delegationBuilder := builder.NewDelegationOutputBuilderFromPrevious(prevOutput).EndEpoch(delegationEndEpoch)
+	if prevOutput.DelegationID == iotago.EmptyDelegationID() {
+		delegationBuilder.DelegationID(iotago.DelegationIDFromOutputID(input.OutputID()))
+	}
+
+	delegationOutput := delegationBuilder.MustBuild()
+
+	signedTransaction := lo.PanicOnErr(w.createSignedTransactionWithOptions(
+		WithContextInputs(iotago.TxEssenceContextInputs{
+			&iotago.CommitmentInput{
+				CommitmentID: w.Node.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment().MustID(),
+			},
+		}),
+		WithInputs(utxoledger.Outputs{input}),
+		WithOutputs(iotago.Outputs[iotago.Output]{delegationOutput}),
+		WithSlotCreated(creationSlot),
+	))
+
+	// register the outputs in the wallet
+	w.registerOutputs(transactionName, signedTransaction.Transaction)
+
+	return signedTransaction
+}
+
+func (w *Wallet) TransitionAccount(transactionName string, inputName string, opts ...options.Option[builder.AccountOutputBuilder]) *iotago.SignedTransaction {
+	input, exists := w.outputs[inputName]
+	if !exists {
+		panic(fmt.Sprintf("account with alias %s does not exist", inputName))
+	}
+
+	accountOutput, ok := input.Output().Clone().(*iotago.AccountOutput)
+	if !ok {
+		panic(fmt.Sprintf("output with alias %s is not *iotago.AccountOutput", inputName))
+	}
+
+	accountBuilder := builder.NewAccountOutputBuilderFromPrevious(accountOutput)
+	accountOutput = options.Apply(accountBuilder, opts).MustBuild()
+
+	signedTransaction := lo.PanicOnErr(w.createSignedTransactionWithOptions(
+		WithAccountInput(input, true),
+		WithContextInputs(iotago.TxEssenceContextInputs{
+			&iotago.BlockIssuanceCreditInput{
+				AccountID: accountOutput.AccountID,
+			},
+			&iotago.CommitmentInput{
+				CommitmentID: w.Node.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment().MustID(),
+			},
+		}),
+		WithOutputs(iotago.Outputs[iotago.Output]{accountOutput}),
+	))
+
+	// register the outputs in the wallet
+	w.registerOutputs(transactionName, signedTransaction.Transaction)
+
```

Comment:
For the recipient wallet, we have to do it here because we don't pass the recipient wallet itself to `createSignedTransactionWithOptions`. It's replacing the step that the firefly wallet would have to do to check for new outputs received on its addresses, but maybe we could automate this a little more in the wallet, I will see if I can think of something.

Reply:
I moved the registerOutputs for the sending wallet into the `createSignedTransactionWithOptions` at least, but we still need to register them for the recipient wallet explicitly for now.